pkg/scheduler/core/generic_scheduler.go

```diff
@@ -203,27 +205,58 @@ func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, fwk framewor
 
 	// Run "prefilter" plugins.
 	s := fwk.RunPreFilterPlugins(ctx, state, pod)
+	allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
+	if err != nil {
+		return nil, nil, err
+	}
 	if !s.IsSuccess() {
 		if !s.IsUnschedulable() {
 			return nil, nil, s.AsError()
 		}
 		// All nodes will have the same status. Some non trivial refactoring is
 		// needed to avoid this copy.
-		allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
-		if err != nil {
-			return nil, nil, err
-		}
 		for _, n := range allNodes {
 			filteredNodesStatuses[n.Node().Name] = s
 		}
 		return nil, filteredNodesStatuses, nil
 	}
 
-	feasibleNodes, err := g.findNodesThatPassFilters(ctx, fwk, state, pod, filteredNodesStatuses)
+	// The "NominatedNodeName" field is set due to the preemption in the previous scheduling cycle, the node
+	// most likely is still the only candidate for the preemptor pod.
+	// Have a try on this node directly when the field is set and the feature is enabled.
+	preferNominatedNode := len(pod.Status.NominatedNodeName) > 0 && utilfeature.DefaultFeatureGate.Enabled(features.PreferNominatedNode)
+	if preferNominatedNode {
+		nnn := pod.Status.NominatedNodeName
+		nodeInfo, err := g.nodeInfoSnapshot.Get(nnn)
+		if err != nil {
+			return nil, nil, err
+		}
+		node := []*framework.NodeInfo{nodeInfo}
+		feasibleNodes, err := g.findNodesThatPassFilters(ctx, fwk, state, pod, filteredNodesStatuses, node)
+		if err != nil {
+			return nil, nil, err
+		}
+		if len(feasibleNodes) != 0 {
+			feasibleNodes, err = g.findNodesThatPassExtenders(pod, feasibleNodes, filteredNodesStatuses)
+		}
+		if err != nil {
+			return nil, nil, err
+		}
+		// Nominated node passes all the filters, scheduler is good to assign this node to the pod.
+		if len(feasibleNodes) != 0 {
+			return feasibleNodes, filteredNodesStatuses, nil
+		}
+		// Remove the nominated node from node list and continue to evaluate the rest of the nodes in the cluster.
```

Comment:
That is not necessary, in fact it is more expensive to iterate over all nodes to remove the nominated node vs just evaluate the node again.